databases/Assignment2_answers.sql

```diff
@@ -0,0 +1,316 @@
+    SELECT distinct Manufacturer
+    FROM Computer
+    INNER JOIN Product USING Model
+    WHERE Product.Type = pc AND RAM > 1024 AND CD = 56x;
+),
+laptops AS (
+    SELECT distinct Manufactuer
+    FROM Computer
+    INNER JOIN Product USING Model
+    WHERE Product.Type = laptop AND Computer.Screen > 15;
+),
+printers AS (
+    SELECT distinct Manufacturer
+    FROM Printer
+    INNER JOIN Product USING Model
+    WHERE Price < 300;
+)
+SELECT c.Manufacturer
+FROM computers c
+INNER JOIN laptops l USING Manufacturer
+INNER JOIN printers p USING Manufacturer
+;
+
+
+
+
+# do sequential JOINS happen simultaneously?
+SELECT DISTINCT Product.Manufacturer
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = pc AND Computer.RAM > 1024 AND Computer.CD = 56x
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.Screen > 15
+INNER JOIN Printer
+WHERE Printer.Price < 300
+GROUP BY Product.Manufacturer;
+
+
+
+^ ERROR. not how innerjoins work :(
+
+
+
+
+
+
+
+
+
+16. What HD sizes are common to at least 2 computers?
+SELECT HD
+FROM Computer
+GROUP BY HD
+HAVING COUNT(Model) > 2;
+
+
+17. Which manufacturers sell more than one computer that
+has a processor speed of less than 3 GHz?
+SELECT Product.Manufacturer
+FROM Product
+WHERE Computer.Speed < 3
+INNER JOIN Computer USING Model
+GROUP BY Product.Manufacturer
+HAVING COUNT(Product.Manufacturer) > 1;
+
+
+18. Which manufacturers sell the fastest computers?
+# general case where I do not know fastest speed
+SELECT DISTINCT Product.Manufacturer
+FROM Product
+INNER JOIN Computer USING (Model)
+WHERE Speed = max(Speed)
+# ^ verify if this is legit...it does not compile for me
+# instead doing:
+# WHERE Speed = (SELECT MAX(Speed) FROM Computer)
+# works, however this is basically a subquery.....so efficiency????? any way to not subquery?
+GROUP BY Manufacturer;
+
+
+
+# if I know the speed, then I can just do:
+SELECT Product.Manufacturer
+FROM Product
+INNER JOIN Computer
+WHERE Speed
+
+GROUP BY Product.Manufacturer
+HAVING
+
+
+19. Which manufacturers do not sell laptops?
+SELECT DISTINCT Manufacturer
+FROM Product
+EXCEPT
+SELECT DISTINCT Manufacturer
+FROM Product
+WHERE Type = laptop;
+
+# INCORRECT. does not account for case where manufacturer does laptop AND something else.
+SELECT DISTINCT Manufacturer
+FROM Product
+WHERE Type != laptop
+
+
+
+
+20. Which manufacturers offer at least 2 laptop models?
+SELECT Manufacturer
+FROM Product
+WHERE Type = laptop
+GROUP BY Manufacturer
+HAVING COUNT(Model) >= 2;
+
+
+21. Which manufacturers offer exactly 3 types of personal computers?
+SELECT Manufacturer
+FROM Product
+WHERE Type = pc
+GROUP BY Manufacturer
+HAVING COUNT(Model) = 3;
+
+22. You need to buy: # I can intersect this too, probably
+• a computer that has
+– at least 1024 MB of RAM,
+– a processor speed of 3 GHz or 3.5 GHz,
+– a HD of at least 160 GB, and
+• a color printer.
+Which manufacturers can offer you this combination and at what total price?
```

Comment:
Question is kind of ambiguous because there can be multiple such combinations per manufacturer, so there could be multiple prices. Maybe you can list each pair, or the max/min of all of them